contracts/script/PlonkVerifierV2.s.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.8.20;
+
+import { Script } from "forge-std/Script.sol";
+
+import { Options, Upgrades } from "openzeppelin-foundry-upgrades/Upgrades.sol";
+/// @notice Deploys an upgradeable Plonk Verifier Contract using the OpenZeppelin Upgrades plugin.
+/// @dev The Upgrades library has a deployImplementation function which is used here
+
+contract DeployPlonkVerifierV2Script is Script {
+    string public contractName = "PlonkVerifierV2.sol";
+
+    function run() public returns (address contractAddress) {
+        // get the deployer info from the environment and start broadcast as the deployer
+        address deployer;
+        string memory ledgerCommand = vm.envString("USE_HARDWARE_WALLET");
+        if (keccak256(bytes(ledgerCommand)) == keccak256(bytes("true"))) {
+            deployer = vm.envAddress("DEPLOYER_HARDWARE_WALLET_ADDRESS");
```

Comment:
Is the reason for having these scripts in addition to the rust deploy script just that they support using hardware wallets?

Reply:
because it was needed for immediate deploys last week and the alloy code wasn't merged in when i first started - i created foundry scripts to support it